diff --git a/browser/components/BrowserGlue.sys.mjs b/browser/components/BrowserGlue.sys.mjs
index 5dcbb17056..465ff44381 100644
--- a/browser/components/BrowserGlue.sys.mjs
+++ b/browser/components/BrowserGlue.sys.mjs
@@ -1605,7 +1605,8 @@
     // Use an increasing number to keep track of the current state of the user's
     // profile, so we can move data around as needed as the browser evolves.
     // Completely unrelated to the current Firefox release number.
-    const APP_DATA_VERSION = 162;
+    // LW: Ugly hack, remove once APP_DATA_VERSION hits 163 in 147
+    const APP_DATA_VERSION = 162.51;
     const PREF = "browser.migration.version";
 
     let profileDataVersion = Services.prefs.getIntPref(PREF, -1);
diff --git a/browser/components/ProfileDataUpgrader.sys.mjs b/browser/components/ProfileDataUpgrader.sys.mjs
index 2a630b40f6..3829044d2b 100644
--- a/browser/components/ProfileDataUpgrader.sys.mjs
+++ b/browser/components/ProfileDataUpgrader.sys.mjs
@@ -935,6 +935,11 @@
       Services.prefs.setBoolPref("signon.rustMirror.migrationNeeded", true);
     }
 
+    if (existingDataVersion < 162.5) {
+      // LibreWolf: Remove obsolete OCSP pref
+      Services.prefs.clearUserPref("security.OCSP.require");
+    }
+
     // Update the migration version.
     Services.prefs.setIntPref("browser.migration.version", newVersion);
   },
